### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by Lawshiga**
https://docs.google.com/document/d/1ld_YQY7zvyt5Kctgd7U-YHa0wEcg8Mn5uqh4h6ka-Ek/edit

## Automation tests
 - Unit tests 
   > Code coverage information : 90%
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

 
https://github.com/ethereum/wiki/wiki/JSON-RPC

### CODE REVIEW

**Review by BuddhiWathsala — CHANGES_REQUESTED** [review]
Please remove unwanted files 
- `component/.attach_pid6313`
- `docs/about/license.md`
- `docs/api/.md`
- `docs/license.md`

File: component/src/main/java/org/wso2/extension/siddhi/io/ethereum/util/EthereumUtil.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        blockMap.put(EthereumConstants.BLOCK_MINER, block.getMiner());
+        blockMap.put(EthereumConstants.BLOCK_RECEIPT_ROOT, block.getReceiptsRoot());
+        blockMap.put(EthereumConstants.BLOCK_STATE_ROOT, block.getStateRoot());
+        blockMap.put(EthereumConstants.BLOCK_TRANSACTIONS_ROOT, block.getTransactionsRoot());
+        blockMap.put(EthereumConstants.BLOCK_LOGS_BLOOM, block.getLogsBloom());
+        blockMap.put(EthereumConstants.BLOCK_SHA3_UNCLES, block.getSha3Uncles());
+        return blockMap;
+    }
+
+    // to create filtering
```
Comment (BuddhiWathsala) [inline component/src/main/java/org/wso2/extension/siddhi/io/ethereum/util/EthereumUtil.java]: Better to add comments like [this](https://github.com/siddhi-io/siddhi-io-http/blob/master/component/src/main/java/io/siddhi/extension/io/http/source/HttpRequestSource.java#L548)

File: component/src/test/java/org/wso2/extension/siddhi/io/ethereum/source/TestCaseOfEthereumSource.java [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    public void ethereumFilterNewBlockTest1() throws InterruptedException {
+
+        log.info("---------------------------------------------------------------------------------------------");
+        log.info("Ethereum Source Test Case of Filtering new blocks.");
+        log.info("---------------------------------------------------------------------------------------------");
+        SiddhiManager siddhiManager = new SiddhiManager();
+        String siddhiApp = "@App:name('test') \n" +
+                "@source(type='ethereum', \n" +
+                "@map(type='keyvalue'), \n" +
+                "uri = 'ws://localhost:8546', \n" +
```
Comment (niruhan) [inline component/src/test/java/org/wso2/extension/siddhi/io/ethereum/source/TestCaseOfEthereumSource.java]: are you starting ethereum publisher locally by yourself? This test file should be able to run as it is on any machine so if you are starting by yourself please do it programatically inside this test case class

**Review by niruhan — CHANGES_REQUESTED** [review]

File: docs/images/siddhi-logo-w.svg [hunk 3]
```diff
@@ -0,0 +1,163 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<!--
+  ~ Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (minudika) [inline docs/images/siddhi-logo-w.svg]: license should be updated

File: docs/images/siddhi-logo.svg [hunk 4]
```diff
@@ -0,0 +1,19 @@
+<!--
+  ~ Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (minudika) [inline docs/images/siddhi-logo.svg]: the license should be updated. check and fix other places as well

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+                        <goals>
+                            <goal>generate-md-docs</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+    <properties>
+        <siddhi.version>4.5.5</siddhi.version>
```
Comment (minudika) [inline pom.xml]: use the latest siddhi version

File: README.md [hunk 6]
```diff
@@ -1 +1,44 @@
-# siddhi-io-ethereum
\ No newline at end of file
+﻿siddhi-io-ethereum
+======================================
+
+
```
Comment (niveathika) [inline README.md]: Complete readme

File: docs/images/siddhi-logo-w.svg (hunk 3 above)
Comment (niveathika) [inline docs/images/siddhi-logo-w.svg]: License Year, check in other places

File: mkdocs.yml [hunk 7]
```diff
@@ -0,0 +1,16 @@
+site_name: Siddhi-IO-Ethereum
```
Comment (niveathika) [inline mkdocs.yml]: Use new site updated mkdoc file

File: pom.xml [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+                <artifactId>testng</artifactId>
+                <version>${testng.version}</version>
+                <scope>test</scope>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.wso2.carbon.maven</groupId>
```
Comment (niveathika) [inline pom.xml]: Remove unnecessary plugin

**Review by niveathika — CHANGES_REQUESTED** [review]

File: component/src/test/java/org/wso2/extension/siddhi/io/ethereum/source/TestCaseOfEthereumSource.java [hunk 9]
```diff
@@ -0,0 +1,274 @@
...
+                for (Event event : events) {
+                    eventCount.getAndIncrement();
+                    eventArrived.set(true);
+                }
+            }
+        });
+        siddhiAppRuntime.start();
+        SiddhiTestHelper.waitForEvents(waitTime, 2, eventCount, timeout);
+
+        log.info("pause");
```
Comment (minudika) [inline component/src/test/java/org/wso2/extension/siddhi/io/ethereum/source/TestCaseOfEthereumSource.java]: Please put some context to these logs..

**Review by minudika — CHANGES_REQUESTED** [review]

File: component/src/main/java/org/wso2/extension/siddhi/io/ethereum/util/EthereumUtil.java [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+        blockMap.put(EthereumConstants.BLOCK_TRANSACTIONS_ROOT, block.getTransactionsRoot());
+        blockMap.put(EthereumConstants.BLOCK_LOGS_BLOOM, block.getLogsBloom());
+        blockMap.put(EthereumConstants.BLOCK_SHA3_UNCLES, block.getSha3Uncles());
+        return blockMap;
+    }
+
+    // called to create filtering
+    public void createFilter(String filterName, String fromBlock, String toBlock,
+                             SourceEventListener sourceEventListener, Web3j web3j) {
+
```
Comment (ramindu90) [inline component/src/main/java/org/wso2/extension/siddhi/io/ethereum/util/EthereumUtil.java]: lets remove these unnecessary spaces on top of each method

**Review by mohanvive — APPROVED** [review]

**Pull request merged**